src/Tools/ExternalAccess/Razor/RazorCSharpProximityExpressionResolver.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp.Debugging;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    internal class RazorCSharpProximityExpressionResolver
+    {
+        public IList<string> GetProximityExpressions(SyntaxTree syntaxTree, int absoluteIndex)
+            => CSharpProximityExpressionsService.Do(syntaxTree, absoluteIndex);
```

Comment:
❗ `Do` is a private helper method. I'm going to send a separate pull request to change the accessibility to `private` and expose it to testing through a test accessor.

Reply:
WOuld you like me to hold off on this PR until then or do it in parallel?